src/safeds/ml/nn/_model.py

```diff
@@ -0,0 +1,250 @@
+import copy
+from typing import Self
+
+import numpy as np
+import torch
+from torch import nn
+
+from safeds.data.tabular.containers import Column, Table, TaggedTable
+from safeds.exceptions import ClosedBound, ModelNotFittedError, OutOfBoundsError
+from safeds.ml.nn._fnn_layer import FNNLayer
+
+
+class RegressionNeuralNetwork:
+    def __init__(self, layers: list):
+        self._model = _PytorchModel(layers, is_for_classification=False)
+        self._batch_size = 1
+        self._is_fitted = False
+
+    def fit(self, train_data: TaggedTable, epoch_size: int = 25, batch_size: int = 1) -> Self:
```

Comment:
* Add callback `on_epoch_completion` callback that gets called whenever an epoch is done.
* Add callback `on_batch_completion` callback that gets called whenever a batch is done.